### PR TITLE
Fix disconnected port handling

### LINKED ISSUE #1433: Mode transition fails to trigger in some
I found a simple modal model where a transition fails to trigger when it should.  This is checked into the [transition-trigger branch](https://github.com/lf-lang/lingua-franca/tree/transition-trigger) here:

https://github.com/lf-lang/lingua-franca/blob/transition-trigger/test/C/src/modal_models/TransitionTrigger.lf

Even stranger, if you uncomment the empty reaction on [line 24](https://github.com/lf-lang/lingua-franca/blob/transition-trigger/test/C/src/modal_models/TransitionTrigger.lf#L24), then the test passes. This reaction should really have no effect.

### PR DESCRIPTION
**Pull request by edwardalee**
This fixes #1433 and #1489.

### CODE REVIEW

**Comment by lhstrh** [comment]
Is there anything material to the fix in the reactor-c update?

**Comment by edwardalee** [comment]
> Is there anything material to the fix in the reactor-c update?

No, but I'm testing with reactor-c/main, so it needs to be aligned.

File: test/C/src/TriggerIssue.lf [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/** Destination cannot react to y. Only if x is also a trigger (uncomment reaction) it will work. */
```
Comment (lhstrh) [inline test/C/src/TriggerIssue.lf:1]: This comment seem wrong? What is the test meant to capture? The output `x` is never set, but it's also not connected. An output is produced on `y`, and the downstream reactor has a reaction that is triggered by it. Did this not work prior to the change?

Comment (edwardalee) [reply test/C/src/TriggerIssue.lf]: Good catch!  This comment was left over from the issue.  Fixed in d6d6cd6deee29db19ec08812f37cd67628bb5c6f.

**Review by lhstrh — APPROVED** [review]
LGTM!

**Pull request merged**